### PR TITLE
Modularize refactor

### PR DESCRIPTION
**Pull request by alexmario74**
The purpose of this changes proposal is to introduce some code organization.

First the function wrapper that isolate the code from the global scope.

Then the use of arrow functions.

Avoiding the use of this even with different pointers (self).

### CODE REVIEW

File: js/app.js [hunk 1]
```diff
@@ -1,262 +1,269 @@
...
+
+    const parseHeaders = headers =>
+      headers.trim().split('\n')
+        .map(header =>
+          header.split(':')
+            .map(h => h.trim()))
+        .map(([name, value]) => ({ name, value }));
+
+    const convertToHeaderObj = headersList =>
+      headersList.reduce((acc, [name, value]) => {
```
Comment (mirkoperillo) [inline js/app.js]: If I put some header in the request this method invocation throws an exception
TypeError: (destructured parameter) is not iterable.

Same problem with methods:
* convertToUrlEncoded
* convertToFormData

**Comment by alexmario74** [comment]
It looks like the firefox don't support the destructuring.

I changed the code, I just pushed on my fork, hope this will be included in the pull request.

Let me know if you find any other issue.

File: js/app.js [hunk 2]
```diff
@@ -1,262 +1,269 @@
...
-    localforage.iterate(function(value,key, iterationNumber) {
-      self.bookmarks.push(JSON.parse(value));
+(($, localforage, ko) => {
+  const makeBookmark = (id, request, name) => ({ id, request, name });
+
+  const makeRequest = (method, url, headers, bodyType, body) =>
+    ({ method, url, headers, bodyType, body });
+
+  function AppViewModel() {
+    const Resting = {
```
Comment (mirkoperillo) [inline js/app.js]: my curiosity: what is the idea behind collecting all view fields ins object ?

Comment (alexmario74) [reply js/app.js]: There are several reasons.

The first is the issue with _this_ in Javascript. As you can change the scope of any object any time, you could have attributes of method rewritten.

Such issues are very complicated to debug.

And that is connected to a design concern: encapsulation.

If you create an object to return with all the accessible fields and method, you can hide the stuff you don't want some developer access or change.

It is called "Revealed Module Pattern", here you can find a better documentation: https://scotch.io/bar-talk/4-javascript-design-patterns-you-should-know

**Comment by mirkoperillo** [comment]
@alexmario74  thank you for your contribution, I have learned a lot analyzing it

**Pull request merged**